install/update_0831_084.php

```diff
@@ -212,25 +236,30 @@ function update0831to084() {
    $DB->query("SET SESSION group_concat_max_len = 4194304;");
    foreach ($changes as $ruletype => $field) {
       // Get rules
-      $query = "SELECT GROUP_CONCAT(`id`)
-                FROM `glpi_rules`
-                WHERE `sub_type` = '".$ruletype."'
-                GROUP BY `sub_type`";
-      if ($result = $DB->query($query)) {
-         if ($DB->numrows($result)>0) {
-            // Get rule string
-            $rules = $DB->result($result, 0, 0);
-
-            // Update actions
-            foreach ($status as $old => $new) {
-               $query = "UPDATE `glpi_ruleactions`
-                         SET `value` = '$new'
-                         WHERE `field` = '$field'
-                               AND `value` = '$old'
-                               AND `rules_id` IN ($rules)";
-
-               $DB->queryOrDie($query, "0.84 update datas for rules actions");
-            }
+      $rulesIterator = $DB->request([
+         'SELECT'    => new \QueryExpression(
+            "GROUP_CONCAT(" . DBmysql::quoteName("id") . ") AS ids"
+         ),
+         'FROM'      => "glpi_rules",
+         'WHERE'     => ['sub_type' => $ruletype],
+         'GROUPBY'   => 'sub_type'
+      ]);
+
+      if (count($rulesIterator)) {
+         // Get rule string
+         $rules = $rulesIterator->next()['ids'];
+
+         // Update actions
+         foreach ($status as $old => $new) {
+            $DB->updateOrDie('glpi_ruleactions', [
+                  'value' => $new
+               ], [
+                  'field'     => $field,
+                  'value'     => $old,
+                  'rules_id'  => explode(",", $rules)
```

Comment:
Not sure we ant implode here

Reply:
I think we need an array to get an "IN => (a,b,c,...)" criteria, is there a better way to do this ?

Reply:
Yes indeed, an array is needed. I did not pay attention you get a string with commas from the DB, so this is ok